test/html_parser.js

```diff
@@ -140,5 +140,12 @@ module.exports.toString = function(assert) {
     assert.ok(doc.toString({ type: 'xml' }).indexOf('<?xml') > -1);
     assert.ok(doc.toString({ type: 'xhtml' }).indexOf('<?xml') > -1);
     assert.ok(doc.toString({ type: 'xml', selfCloseEmpty:true }).indexOf('<a/>') > -1);
+
+    doc = libxml.parseHtml("<a>Something&nbsp;with a space</a>");
+    assert.ok(doc.toString().indexOf('<?xml') === -1);
+    assert.ok(doc.toString({ type: 'xml' }).indexOf('<?xml') > -1);
+    assert.ok(doc.toString({ type: 'xhtml' }).indexOf('<?xml') > -1);
+    assert.ok(doc.toString({ type: 'xhtml' }).indexOf('&nbsp;') === -1);
+    assert.ok(doc.toString({ type: 'xhtml', encoding: 'HTML' }).indexOf('&nbsp;') > -1);
```

Comment:
What does `encoding: 'HTML'` actually do? Isn't this supposed to be a character encoding?